### PR TITLE
LoRa basic station PoC

### PR DESCRIPTION
**Pull request by jpm-canonical**
Using a docker companion snap, the container is started by the `command` and stopped by `stop-command`.

Log output looks like this when it is successfully running:
```
...
2024-08-15T14:17:46Z lora-basicstation.lora-basicstation[17038]: 2024-08-15 14:17:46.203 [S00:INFO] Concentrator started (4s856ms)
2024-08-15T14:18:28Z lora-basicstation.lora-basicstation[17038]: 2024-08-15 14:18:28.132 [SYN:INFO] MCU/SX130X drift stats: min: +0.0ppm  q50: +0.5ppm  q80: +1.0ppm  max: -7.6ppm - threshold q90: -2.4ppm
2024-08-15T14:18:28Z lora-basicstation.lora-basicstation[17038]: 2024-08-15 14:18:28.132 [SYN:INFO] Mean MCU drift vs SX130X#0: -0.6ppm
...
```
And when it is stopped we see this:
```
2024-08-15T14:18:32Z systemd[1]: Stopping snap.lora-basicstation.lora-basicstation.service - Service for snap application lora-basicstation.lora-basicstation...
2024-08-15T14:18:43Z lora-basicstation.lora-basicstation[17402]: lora-basicstation
2024-08-15T14:18:43Z systemd[1]: snap.lora-basicstation.lora-basicstation.service: Main process exited, code=exited, status=137/n/a
2024-08-15T14:18:43Z lora-basicstation.lora-basicstation[17491]: lora-basicstation
2024-08-15T14:18:43Z systemd[1]: snap.lora-basicstation.lora-basicstation.service: Failed with result 'exit-code'.
2024-08-15T14:18:43Z systemd[1]: Stopped snap.lora-basicstation.lora-basicstation.service - Service for snap application lora-basicstation.lora-basicstation.
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+sudo snap set lora-basicstation env.tc-key=NNSXS...
+```
+
+Or with a single command:
+
+```
+sudo snap set lora-basicstation env.gateway-eui=DEADFFFEBEEF env.model=SX1301 \
+env.tc-key=NNSXS... env.tc-uri=wss://eu1.cloud.thethings.network:8887 \
+env.device=/dev/spidev0.0 env.interface=SPI env.spi-speed=200000 \
+env.use-libgpiod=1 env.gpio-chip=gpiochip4 env.reset-gpio=22
```
Comment (farshidtz) [inline README.md:44]: Hard to read. How about one key-value per line?

```suggestion
sudo snap set \
    lora-basicstation env.gateway-eui=DEADFFFEBEEF \
    env.model=SX1301 \
    env.tc-key=NNSXS... env.tc-uri=wss://eu1.cloud.thethings.network:8887 \
    ...
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+sudo docker stop lora-basicstation
+sudo docker rm -f lora-basicstation
+```
+
+### Reset concentrator
+
+To manually reset the gateway concentrator hardware, you can toggle the reset GPIO pin. A conveniance script is provided to assist with this. The following example is for the Raspberry Pi 5. Replace `RESET_GPIO="22"` with the correct GPIO your concentrator is connected to. If you are using an older Raspberry Pi, set `GPIO_CHIP` to `gpiochip0`.
+
+```
+sudo GPIO_CHIP="gpiochip4" RESET_GPIO="22" POWER_EN_GPIO=0 POWER_EN_LOGIC=0 ./reset.gpiod.sh
```
Comment (farshidtz) [inline README.md:95]: Can this be packages in the same snap?

**Review by farshidtz — APPROVED** [review]
Looks good. I've added some minor comments.

File: README.md (hunk 2 above)
Comment (jpm-canonical) [reply README.md:95]: This reset script, along with a couple other are included in the docker container. We could expose them as separate apps in the snap. I'd rather leave that for a future task.

File: snap/local/bin/run.sh [hunk 3]
```diff
@@ -8,9 +8,7 @@ docker create \
   --name "$container_name" \
   --log-driver none \
   --env-file "$SNAP_COMMON/conf.env" \
-  --device=/dev/spidev0.0:/dev/spidev0.0 \
-  --device=/dev/gpiochip0:/dev/gpiochip0 \
-  --device=/dev/gpiochip4:/dev/gpiochip4 \
+  --privileged \
```
Comment (farshidtz) [inline snap/local/bin/run.sh:11]:
```suggestion
  --device=/dev/$SPI:/dev/$SPI \
  --device=/dev/$GPIO:/dev/$GPIO \
```

Repurpose the existing envs for this.

Set defaults from an install hook. E.g. https://github.com/canonical/openthread-border-router-snap/blob/main/snap/hooks/install

**Pull request merged**